### PR TITLE
Add bypass around IP checks for specific endpoints

### PR DESCRIPTION
**Pull request by ksclarke**
* Upgrade CI to use JDK 17
* Make small syntax tweaks to .gitignore
* Simplify README.md and pom.xml
* Update project dependencies
* Fix logging so container logs [SERV-582]
* Change container's JDK to variable instead of hard-coding
* Add IP check bypass for specific endpoints per Apps Team
* Update tests to use newer `Future` syntax
* Catch failed test exceptions with `aContext.verify(() -> {});`
* Remove file logging config and add `testLogLevel` config

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -615,6 +621,14 @@
...
                 <jvmArg>-DHTTP_PORT=${http.test.port}</jvmArg>
               </jvmArgs>
+              <runArgs>
+                <runArg>-Dlibcal.client1.id="${libcal.client1.id}"</runArg>
+                <runArg>-Dlibcal.client1.secret="${libcal.client1.secret}"</runArg>
+                <runArg>-Dlibcal.client2.id="${libcal.client2.id}"</runArg>
+                <runArg>-Dlibcal.client2.secret="${libcal.client2.secret}"</runArg>
+                <runArg>-Dlibcal.token.endpoint="${libcal.token.endpoint}"</runArg>
+                <runArg>-Dlibcal.base.url="${libcal.base.url}"</runArg>
+              </runArgs>
```
Comment (markmatney) [inline pom.xml:631]: Why this change?

File: README.md [hunk 2]
```diff
@@ -42,9 +43,6 @@ You can also run the application directly on the host machine by defining enviro
     LIBCAL_CLIENT2_SECRET=0123456789abcdef0123456789abcdef \
     LIBCAL_TOKEN_ENDPOINT=https://calendar.library.ucla.edu/1.1/oauth/token \
     LIBCAL_BASE_URL=https://calendar.library.ucla.edu \
-    LIBCAL_AUTH_RETRY_COUNT=3 \
-    LIBCAL_AUTH_RETRY_DELAY=10 \
-    LIBCAL_AUTH_EXPIRES_IN_PADDING=300 \
```
Comment (markmatney) [inline README.md:47]: Even if we eventually change how the app is configured in this regard (which I am on board with), at this point these values are still configurable, so I don't see why this should be removed yet. For example maybe a dev wants to run a local instance with these values tweaked in order to manually test the authentication process (which I've done in the past).

Comment (ksclarke) [reply README.md:47]: Maybe mentioning them in the text below as optional arguments would be better? The way that it is currently makes them look required.

File: pom.xml (hunk 1 above)
Comment (ksclarke) [reply pom.xml:631]: If one wants to run the app using the Maven Vert.x plugin in what we've called "live" mode in the past. I used that a bit when trying to shortcut the long build cycle of the Docker container.

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]
